### DISCUSSION TITLE
Happy to help with lingua-js!

### DISCUSSION THREAD

**Discussion by lhr0909** [pemistahl/lingua-rs#110] [General]
Hey Peter,

I need to use lingua again in a JS project, and I haven't added Node v18 support yet this year. And then I turned into the main lingua lib and I noticed that you are adding wasm support for lingua and I am happy to take on initial work for the lingua-js package!

Cheers,
Simon

**Comment by pemistahl** [comment]
Hi @lhr0909, thank you for volunteering and sorry for my late response. Even though I encourage everyone to participate, in this case I need to make some necessary adjustments to the [WASM wrapper](https://github.com/pemistahl/lingua-rs/blob/main/src/wasm.rs) first before I can create the lingua-js package. The WASM API should strictly follow the builder pattern as in the Rust API but I did not know how to accomplish this back then. While working on the WASM API for [grex](https://github.com/pemistahl/grex/blob/main/src/wasm.rs), I found out how to do it. This has to be applied to lingua-rs as well.

I don't know whether I was clear enough about this. If so, I very much appreciate a PR with the necessary changes. Thank you. :)

**Reply by pemistahl** [reply]
I've opened issue #122 for this.

**Reply by lhr0909** [reply]
Cool. @pemistahl I will follow closely on this and if I have more time in my hand, will definitely help you with #122 as well as the lingua-js package. I used the Python lingua package for the script I needed, and I am investing my time and work towards the NLP world (been fighting the tensors in my head lol). Once the builder pattern is added I can definitely follow your wasm builds to set up lingua-js for you!
